src/debugger/debugging.ts

```diff
@@ -0,0 +1,88 @@
+import * as vscode from 'vscode';
+import {
+    doCheckModel, getSpecFiles
+} from '../commands/checkModel';
+
+export const TLAPLUS_DEBUG_LAUNCH_CHECKNDEBUG = 'tlaplus.debug.checkAndDebugEditorContents';
+export const TLAPLUS_DEBUG_LAUNCH_DEBUG = 'tlaplus.debug.debugEditorContents';
+
+export class TLADebugAdapterServerDescriptorFactory implements vscode.DebugAdapterDescriptorFactory {
+
+    createDebugAdapterDescriptor(session: vscode.DebugSession, executable: vscode.DebugAdapterExecutable | undefined):
+        vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
+        return new vscode.DebugAdapterServer(session.configuration['port']);
+    }
+}
+
+/**
+ * Attaches the DAP front-end to an already running TLC debugger.
+ */
+export async function debugSpec(
+    resource: vscode.Uri | undefined,
+    diagnostic: vscode.DiagnosticCollection,
+    context: vscode.ExtensionContext
+): Promise<void> {
+    let targetResource = resource;
+    if (!targetResource && vscode.window.activeTextEditor) {
+        // Since this command is registered as a button on the editor menu, I don't
+        // think this branch is ever taken.  It's here because the DAP example has it.
+        targetResource = vscode.window.activeTextEditor.document.uri;
+    }
+    if (targetResource) {
+        // Attaching to a separately launched TLC leaves the result view (webview) empty.
+        // However, TLC sends its output via the DAP Output event
+        // (https://microsoft.github.io/debug-adapter-protocol/specification#Events_Output)
+        // to VSCode.  Somebody has to figure out how to wire this up. In the meantime,
+        // users have to manually "parse" the output on VSCode's DebugConsole. Unfortunately,
+        // it is TLC's '-tool' output containing "@!@!@" markers around each message.
+        vscode.debug.startDebugging(undefined, {
+            type: 'tlaplus',
+            name: 'Debug Spec',
+            request: 'launch',
+            program: targetResource.fsPath,
+            port: '4712'
```

Comment:
Since TLC allows to redefine this default port, I think this number should be a setting.

Reply:
I can't think of a use case where users would want to change the port when they run the debugger from VSCode.  When the TLA+ debugger runs externally, users should enter the port via the command prompt, not as a setting.

Reply:
What if I manually run TLC with `debugger port=4713` and want to attach to it from VSCode?

Reply:
There is the other command ~"attach to running TLC debugger" that should eventually read the port from the new prompt implemented by @klinvill.  Right now, this command expects TLC to listen on its default port, which likely takes care of 83.234827% of all users. :-) Also, a setting wouldn't work well if a user wants to connect to two or more instances of TLC running concurrently.

Reply:
If this "Debug a running model" command that uses this default port will be eventually replaced by "Attach to running TLC debugger" with a port number prompt, than it's OK, thanks.